_posts/2025-10-08-#33300.md

```diff
@@ -10,17 +10,66 @@ status: upcoming
+
+- See our [fuzzing docs](https://github.com/bitcoin/bitcoin/blob/master/doc/fuzzing.md) to start fuzzing with libFuzzer and AFL++.
+
+- Check out this PR's [test coverage](https://marcofleon.github.io/coverage/cmpctblock/) (`blockencodings.cpp` and `net_processing.cpp` are probably most useful to look at). Walking through a coverage report to see which branches are hit or not hit is a good way to start evaluating a fuzz harness.
+
+- An effective fuzz test will be deterministic, as this makes bugs reproducible. If you’re able to get the AFL++ fuzzer running, you will see a stability metric. The goal is to get to 100%, perfect determinism! After gathering some inputs, you can also run the [fuzz determinism script](https://github.com/bitcoin/bitcoin/blob/master/contrib/devtools/deterministic-fuzz-coverage/src/main.rs) over the corpus to check for potential sources of instability.
+
+
+## Fun Exercise
+
+
+- If you get the `cmpctblock` harness running, revert PR [#33296](https://github.com/bitcoin/bitcoin/pull/33296) and see if you can reproduce the crash on this [`Assume`](https://github.com/bitcoin/bitcoin/blob/689a32197638e92995dd8eb071425715f5fdc3a4/src/net_processing.cpp#L3333) statement.
+
+- Yeah I know, fuzzing is cool.
 
 
 ## Questions
 
-1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
--->
+
+1. Did you review the PR? Concept ACK, approach ACK, tested ACK, or NACK? What was your review approach? Were you able to get the fuzz test running?
+
+2. Where in the codebase are the main compact block helpers and processing logic? Name some of the relevant classes and functions. (Hint: nothing like a little search for `NetMsgType::CMPCTBLOCK` to get you started)
+
+3. The fuzz test sends `SENDCMPCT` messages with `high_bandwidth` randomly set. What is the difference between high bandwidth and low bandwidth compact block relay? How many high bandwidth peers are allowed and does the fuzz harness test this limit?
+
+4. Compare `-testdatadir` and the new `-fuzzcopydatadir`. Why is the latter useful for performance, and why isn’t a fresh `TestingSetup` that mines blocks on each iteration acceptable here?
+
+5. Look at `create_block` in the harness. How many transactions do the generated blocks contain, and where do they come from? What compact block scenarios might be missed with only a few transactions in a block?
+
+6. In commit [ecba804](https://github.com/bitcoin/bitcoin/pull/33300/commits/ecba804b0d034bbb6e61510cd2161b2ae449ac8d) `FinalizeHeader` is moved to `util.h` so it can be used by any fuzz test. What does this function do and why doesn’t it stall the fuzzer?
```

Comment:
I think it would be useful to add a bulletpoint in the notes on stalling: e.g. why it's relevant, how it's measured, what's acceptable, ... - just to give some context.